phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexScrutinyToolPerformanceIT.java

```diff
@@ -0,0 +1,246 @@
+        if (!Strings.isNullOrEmpty(viewDdl)) {
+            createTestTable(getUrl(), String.format(viewDdl, viewFullName, dataTableFullName));
+            createTestTable(getUrl(), String.format(indexTableDdl, indexTableName, viewFullName));
+        } else {
+            createTestTable(getUrl(), String.format(indexTableDdl, indexTableName, dataTableFullName));
+        }
+        props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        conn = DriverManager.getConnection(getUrl(), props);
+        String dataTableUpsert = String.format(UPSERT_SQL, dataTableFullName);
+        dataTableUpsertStmt = conn.prepareStatement(dataTableUpsert);
+        String viewUpsert = String.format(UPSERT_SQL, viewFullName);
+        viewUpsertStmt = conn.prepareStatement(viewUpsert);
+        conn.setAutoCommit(false);
+        testTime = EnvironmentEdgeManager.currentTimeMillis() - 1000;
+
+    }
+
+    @After
+    public void teardown() throws SQLException {
+        if (conn != null) {
+            conn.close();
+        }
+    }
+
+    @Test
+    public void testSkipScan() throws Exception {
+        String baseTableName = dataTableName;
+        PreparedStatement upsertStmt = dataTableUpsertStmt;
+        if (!Strings.isNullOrEmpty(viewDdl)) {
+            baseTableName = viewName;
+            upsertStmt = viewUpsertStmt;
+        }
+
+        for (int i=0; i < 2; i++) {
+            if (dataTableDdl.contains("MULTI_TENANT")) {
+                upsertRowMultiTenant(upsertStmt, "val"+i, i,"name-" + i, 19999);
+            } else {
+                upsertRow(upsertStmt, i, "name-" + i, 19999);
+            }
+            conn.commit();
+        }
+
+        List<Job> completedJobs = IndexScrutinyToolIT.runScrutiny(schemaName, baseTableName,
+                indexTableName, null, SourceTable.DATA_TABLE_SOURCE);
+        PreparedStatement targetStmt = IndexScrutinyTool.getTargetTableQueryForScrutinyTool();
+        assertSkipScanFilter(targetStmt);
+        Job job = completedJobs.get(0);
+        Counters counters = job.getCounters();
+        assertEquals(2, getCounterValue(counters, VALID_ROW_COUNT));
+        assertEquals(0, getCounterValue(counters, INVALID_ROW_COUNT));
+
+        completedJobs = IndexScrutinyToolIT.runScrutiny(schemaName, baseTableName,
+                indexTableName, null, SourceTable.INDEX_TABLE_SOURCE);
+        targetStmt = IndexScrutinyTool.getTargetTableQueryForScrutinyTool();
+        assertSkipScanFilter(targetStmt);
+
+        job = completedJobs.get(0);
+        counters = job.getCounters();
+        assertEquals(2, getCounterValue(counters, VALID_ROW_COUNT));
+        assertEquals(0, getCounterValue(counters, INVALID_ROW_COUNT));
+    }
+
+    private void assertSkipScanFilter(PreparedStatement targetStmt) throws SQLException {
+        QueryPlan plan = targetStmt.unwrap(PhoenixStatement.class).getQueryPlan();
+        Scan scan = plan.getScans().get(0).get(0);
+        Filter filter = scan.getFilter();
```

Comment:
Is the presence of a SkipScan filter sufficient to guarantee correctness, or is there some other additional information we need, like what's being filtered?